### PR TITLE
Add https://www.npmjs.com/package/animals types

### PR DESCRIPTION
**Pull request by bencooper222**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [X] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [X] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [X] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [X] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by javiergarzac — APPROVED** [review]

File: types/animals/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// Type definitions for animals 0.0
+// Project: https://github.com/boennemann/animals
+// Definitions by: Ben Cooper <https://github.com/bencooper222>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function main(): string;
+declare namespace main {
+    const words: string[];
+}
+export = main;
```
Comment (peterblazejewicz) [inline types/animals/index.d.ts:10]: this works, but looks like direct portamento from Java,
try this:
```suggestion
declare const animals: {
    (): string;
    readonly words: string[];
}

export = animals;
```

Comment (bencooper222) [reply types/animals/index.d.ts]: Done

**Review by javiergarzac — APPROVED** [review]

File: types/animals/animals-tests.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import animals = require('animals');
+
+
```
Comment (peterblazejewicz) [inline types/animals/animals-tests.ts:3]:
```suggestion
```

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@bencooper222 thanks!

**Comment by bencooper222** [comment]
Ready to merge

**Pull request merged**